### PR TITLE
Add row column screen

### PR DESCRIPTION
**Pull request by Keith-Go**
## Ticket
- Row & Column screen: [Row & Column screen](https://www.notion.so/42297babdb6541c586c45340529b23f3?v=180009363b23459eb8233bf6052d4758&p=e20b3c34d34b4650a5faf1d8d020d1c1&pm=s)

## Video

https://github.com/luke-nguyen-goldenowl/training_keith_dart_challenge/assets/152830147/d9729bd9-534d-4a1d-9ce7-fb521fe90d75

### CODE REVIEW

File: gallery_app/lib/widgets/row_column_expanded_widgets/row_column_menu.dart [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+              Row(
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                children: [
+                  const Text(
+                    "mainAxisSize",
+                    style: TextStyle(fontSize: 18),
+                  ),
+                  DropdownButton(
+                    value: state.mainAxisSize,
+                    items: [MainAxisSize.max, MainAxisSize.min].map((value) {
```
Comment (luke-nguyen-goldenowl) [inline gallery_app/lib/widgets/row_column_expanded_widgets/row_column_menu.dart]: Use MainAxisSize.values => It returns a list of values contained in an enum type

File: gallery_app/lib/widgets/row_column_expanded_widgets/row_column_menu.dart [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                children: [
+                  const Text(
+                    "mainAxisAlignment",
+                    style: TextStyle(fontSize: 18),
+                  ),
+                  DropdownButton(
+                    value: state.mainAxisAlignment,
+                    items: [
+                      MainAxisAlignment.center,
```
Comment (luke-nguyen-goldenowl) [inline gallery_app/lib/widgets/row_column_expanded_widgets/row_column_menu.dart]: Same above

File: gallery_app/lib/widgets/row_column_expanded_widgets/row_column_menu.dart [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                children: [
+                  const Text(
+                    "crossAxisAlignment",
+                    style: TextStyle(fontSize: 18),
+                  ),
+                  DropdownButton(
+                    value: state.crossAxisAlignment,
+                    items: [
+                      CrossAxisAlignment.center,
```
Comment (luke-nguyen-goldenowl) [inline gallery_app/lib/widgets/row_column_expanded_widgets/row_column_menu.dart]: same above

File: gallery_app/lib/widgets/row_column_expanded_widgets/row_column_menu.dart [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+              Row(
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                children: [
+                  const Text(
+                    "verticalDirection",
+                    style: TextStyle(fontSize: 18),
+                  ),
+                  DropdownButton(
+                    value: state.verticalDirection,
+                    items: [VerticalDirection.down, VerticalDirection.up]
```
Comment (luke-nguyen-goldenowl) [inline gallery_app/lib/widgets/row_column_expanded_widgets/row_column_menu.dart]: same above

File: gallery_app/lib/widgets/row_column_expanded_widgets/row_column_menu.dart [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+              Row(
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                children: [
+                  const Text(
+                    "textDirection",
+                    style: TextStyle(fontSize: 18),
+                  ),
+                  DropdownButton(
+                    value: state.textDirection,
+                    items: [TextDirection.ltr, TextDirection.rtl].map((value) {
```
Comment (luke-nguyen-goldenowl) [inline gallery_app/lib/widgets/row_column_expanded_widgets/row_column_menu.dart]: same above

File: gallery_app/lib/widgets/row_column_expanded_widgets/row_column_menu.dart [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+              Row(
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                children: [
+                  const Text(
+                    "textBaseline",
+                    style: TextStyle(fontSize: 18),
+                  ),
+                  DropdownButton(
+                    value: state.textBaseline,
+                    items: [TextBaseline.ideographic, TextBaseline.alphabetic]
```
Comment (luke-nguyen-goldenowl) [inline gallery_app/lib/widgets/row_column_expanded_widgets/row_column_menu.dart]: same above

**Comment by Keith-Go** [comment]
resolved

**Comment by luke-nguyen-goldenowl** [comment]
Approved

File: gallery_app/lib/widgets/row_column_expanded_widgets/row_column_menu.dart [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+                  const Text(
+                    "mainAxisAlignment",
+                    style: TextStyle(fontSize: 18),
+                  ),
+                  DropdownButton(
+                    value: state.mainAxisAlignment,
+                    items: MainAxisAlignment.values.map((value) {
+                      return DropdownMenuItem(
+                        value: value,
+                        child: Text(value.toString().split('.')[1]),
```
Comment (luke-nguyen-goldenowl) [inline gallery_app/lib/widgets/row_column_expanded_widgets/row_column_menu.dart]: Please take note, enum variables have a default getter to return returns its name, for example:
> MainAxisAlignment.spaceBetween.name => "spaceBetween"

**Review by luke-nguyen-goldenowl — APPROVED** [review]

**Pull request merged**